pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -413,27 +413,27 @@ func buildAutoMtlsSettings(
 	meshExternal bool,
 	serviceMTLSMode model.MutualTLSMode) (*networking.ClientTLSSettings, mtlsContextType) {
 	if tls != nil {
+		if tls.Mode == networking.ClientTLSSettings_DISABLE || tls.Mode == networking.ClientTLSSettings_SIMPLE {
+			return tls, userSupplied
```

Comment:
Isn't this duplicate with https://github.com/istio/istio/pull/33292/files#diff-e3b667b806175fc66cb679217641e1638639507c364df2432a60c1d4e87a9739R435

Reply:
It is not because for SIMPLE and when it has metadata certs it builds MUTUAL TLS context and returns. Which is not correct that is what this PR is trying to fix

Reply:
IC